### ISSUE TITLE
duplicated definition error when tags: tag, Tag

### ISSUE THREAD

**Issue by kounoike** [getnikola/nikola#2325]
I write two or more post with case-mismatched tags.
ex. `nikola new_post -t test1 --tags tag` and `$ nikola new_post -t test2 --tags Tag`.

nikola reports this error:
```
$ nikola build
Scanning posts......done!
ERROR: Task generation 'render_site' has duplicated definition of 'render_tags:output/categories/tag.xml'
```

**Comment by Kwpolska** [comment]
This should crash, but with a more helpful message. Nikola version?

**Comment by ralsina** [comment]
@Kwpolska happens in master, anyway.

We do have checks for tags-that-produce-the-same-slug, wonder why it's not triggered.

**Comment by Kwpolska** [comment]
git bisect to the rescue: 361ca292b0a06f864e5c434cf5bcaf7ecaf83551 is the first bad commit.

```
commit 361ca292b0a06f864e5c434cf5bcaf7ecaf83551
Author: Felix Fontein <felix@fontein.de>
Date:   Sun Mar 6 18:14:23 2016 +0100

    First try to pass language to slugify and unslugify (see #2273).

:040000 040000 1dc830befe67f31dd7e307c1b229d047e8ab2040 c1577d31261d20f590448d7cb6809b217db3e8ef M      docs
:040000 040000 f194167f0a46fa4ca954e24621dfd8c62ce53f3b bee8f54413b1a8798944288c5d3b247a71eba728 M      nikola
:040000 040000 d355f83bd25af045d0c63689ec9eacdb124f562b b3d37e598ed66b2ad83e8f69c40fa4fd59d9b66f M      tests
```

All this (wasteful!) Deutsche logic for pretty äöü things ruined an otherwise good function. I’ll try to fix this.

**Comment by Kwpolska** [comment]
The bug was caused on `posts_per_tag` being filled before the loop. I merged the two loops in the correct order and it works now.

Also, it looks like it fixed #2315 — sets were used all along, and this broke something along the way.
